core/primitives/src/shard_layout.rs

```diff
@@ -270,7 +272,7 @@ fn is_top_level_account(top_account: &AccountId, account: &AccountId) -> bool {
 }
 
 /// ShardUId is an unique representation for shards from different shard layout
-#[derive(Hash, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(BorshSerialize, BorshDeserialize, Hash, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
we use `to_bytes` method for `FlatState` key, so let's keep it consistent and use it for all flat state columns instead of borsh serialization